test/gnosisSafePersonalEdition.js

```diff
@@ -49,10 +53,61 @@ contract('GnosisSafePersonalEdition', function(accounts) {
         let executorDiff = await web3.eth.getBalance(executor) - executorBalance
         console.log("    Executor earned " + web3.fromWei(executorDiff, 'ether') + " ETH")
         assert.ok(executorDiff > 0)
-    })
+    });
+
+    it('should deposit and withdraw 1 ETH paying with token', async () => {
+        let token = await safeUtils.deployToken(accounts[0]);
+        let executorBalance = (await token.balances(executor)).toNumber();
+        await token.transfer(gnosisSafe.address, 10000000, {from: accounts[0]});
+        await web3.eth.sendTransaction({from: accounts[0], to: gnosisSafe.address, value: web3.toWei(1.1, 'ether')})
```

Comment:
can we set value to `web3.toWei(1.0, 'ether')` and check at the end of the test that the balance of the safe is 0

Reply:
The gas is paid by the executor and the executor is refunded in token ;)

Also it would be nice to add a third transaction that tries to withdraw 0.5 ether and check that it emits the failure event (see `should deposit and withdraw 1 ETH` test)

Reply:
In the latest commit, I am checking if the balance is 0